From 3d953499227ead08769eb34ed9e7266f90d95de0 Mon Sep 17 00:00:00 2001
From: mdeejay <mdjrussia@gmail.com>
Date: Mon, 3 Jul 2017 01:22:31 +0700
Subject: changes for MTK

Change-Id: Iedd1caa8640c7a22e5fb3b3850820d3f550198df

diff --git a/apps/FMRadio/jni/fmr/libfm_jni.cpp b/apps/FMRadio/jni/fmr/libfm_jni.cpp
index a906887..2949911 100644
--- a/apps/FMRadio/jni/fmr/libfm_jni.cpp
+++ b/apps/FMRadio/jni/fmr/libfm_jni.cpp
@@ -54,7 +54,7 @@ jboolean powerUp(JNIEnv *env, jobject thiz, jfloat freq)
     int tmp_freq;
 
     LOGI("%s, [freq=%d]\n", __func__, (int)freq);
-    tmp_freq = (int)(freq * 10);        //Eg, 87.5 * 10 --> 875
+    tmp_freq = (int)(freq * 100);        //Eg, 87.5 * 10 --> 875
     ret = FMR_pwr_up(g_idx, tmp_freq);
 
     LOGD("%s, [ret=%d]\n", __func__, ret);
