diff -uarN system/netd/server/NatController.cpp system/netd/server/NatController.cpp
--- system/netd/server/NatController.cpp	2019-12-12 18:50:20.235085000 +0700
+++ system/netd/server/NatController.cpp	2019-12-14 16:11:30.513595078 +0700
@@ -375,8 +375,9 @@
         rc = -1;
         goto err_return;
     }
-
-    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add) {
+    // Revert "Don't start tethering if IPv6 RPF is not supported."
+    // STOPSHIP: Make this an error.
+    if (runCmd(ARRAY_SIZE(cmd4), cmd4) && add && false /* STOPSHIP */) {
         rc = -1;
         goto err_rpfilter;
     }
diff -uarN system/netd/server/SoftapController.cpp system/netd/server/SoftapController.cpp
--- system/netd/server/SoftapController.cpp	2019-12-12 18:50:20.239085000 +0700
+++ system/netd/server/SoftapController.cpp	2019-12-14 16:12:39.467644082 +0700
@@ -53,7 +53,9 @@
 #include "wpa_ctrl.h"
 #endif
 
-std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/wlan0"));
+//std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/wlan0"));
+std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/ap0"));
+
 #ifdef LIBWPA_CLIENT_EXISTS
 static const char HOSTAPD_DHCP_DIR[]    = "/data/misc/dhcp";
 #endif
@@ -289,7 +291,8 @@
             "hw_mode=%c\n"
             "ignore_broadcast_ssid=%d\n"
             "wowlan_triggers=any\n",
-            argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+         // argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+            "ap0", argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
 
     std::string fbuf;
     if (argc > 7) {
